tensorflow/python/distribute/cluster_resolver/slurm_cluster_resolver.py

```diff
@@ -27,6 +27,44 @@
 from tensorflow.python.training.server_lib import ClusterSpec
 from tensorflow.python.util.tf_export import tf_export
 
+from hostlist import expand_hostlist, parse_slurm_tasks_per_node
```

Comment:
Can we reimplement `expand_hostlist` and `parse_slurm_tasks_per_node` here? Given the size of TensorFlow as it is, we are trying not to include new packages unless absolutely essential.

I note that `hostlist` is GPL so it is not compatible with TensorFlow's Apache license.

Reply:
Yes could be done, but the format is... special. I guess even though I would not (and did not) look at the source code of hostlist I assume I'll end up with an implementation VERY similar. This could be a license issue too.

I wasn't aware that consuming a GPL Python package from another Apache Python package is not allowed. I mean: It doesn't ship with it. The users installs it.

Reply:
I am not a lawyer, and I am guessing it depends on the interpretation of what "linking" is under GPL, but if calling a function is considered linking, then we may want to avoid this if possible.

https://www.apache.org/licenses/GPL-compatibility.html